libr/core/canal.c

```diff
@@ -5169,3 +5169,112 @@ R_API void r_core_anal_inflags(RCore *core, const char *glob) {
 	r_config_set (core->config, "anal.in", anal_in);
 	free (anal_in);
 }
+
+static bool is_noreturn_function(RCore *core, RAnalFunction *f) {
+	RListIter *iter;
+	RAnalBlock *bb;
+	r_list_foreach (f->bbs, iter, bb) {
+		ut64 pos = bb->addr;
+		RAnalOp op;
+		op.type = R_ANAL_OP_TYPE_ILL;
+
+		// get last opcode
+		while (pos < bb->addr + bb->size) {
```

Comment:
Instead of iterating you could use `r_anal_op` just on the bytes from the address returned by `r_anal_bb_opaddr_i(bb, bb->ninstr - 1)` if you want the last opcode in the basic block.

Reply:
done